libtorrent-rasterbar: broken pkgconfig file

i cannot use [libtorrent-rasterbar](https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/li/libtorrent-rasterbar-2_0_x/package.nix) in a cmake project

`cmake` fails with

```
Imported target "PkgConfig::LIBTORRENT" includes non-existent path

"/nix/store/b19a8ycyd60bcg1fqi0hkh5q01vjc2kg-libtorrent-rasterbar-2.0.11//nix/store/la5036alcfvkcdm2nz6hk68p006i4jwa-libtorrent-rasterbar-2.0.11-dev/include"
```

the problem is in the `libtorrent-rasterbar.pc` file
where `$out` and `$dev` paths are joined like `$out/$dev`

```
$ rg -F rasterbar-2.0.11//nix/store/ /nix/store/la5036alcfvkcdm2nz6hk68p006i4jwa-libtorrent-rasterbar-2.0.11-dev
/nix/store/la5036alcfvkcdm2nz6hk68p006i4jwa-libtorrent-rasterbar-2.0.11-dev/lib/pkgconfig/libtorrent-rasterbar.pc
8:Cflags: -fexceptions -I/nix/store/b19a8ycyd60bcg1fqi0hkh5q01vjc2kg-libtorrent-rasterbar-2.0.11//nix/store/la5036alcfvkcdm2nz6hk68p006i4jwa-libtorrent-rasterbar-2.0.11-dev/include -DTORRENT_LINKING_SHARED ...
```

fixed in https://github.com/milahu/nur-packages/commit/84438f2c3b57f4cffcfe866a6a1c7b8cb91c0df4

sources:

https://github.com/arvidn/libtorrent/blob/RC_2_0/cmake/Modules/GeneratePkgConfig/pkg-config.cmake.in

```
Cflags: @_interface_compile_options@ @_interface_include_dirs@ @_interface_definitions@
```

https://github.com/arvidn/libtorrent/blob/RC_2_0/cmake/Modules/GeneratePkgConfig/generate-pkg-config.cmake.in

when i remove `postPatch`, i get

```
Broken paths found in a .pc file! /nix/store/351afrn507w61bwz3j5lbszbr7c0ssik-libtorrent-rasterbar-2.0.11-dev/lib/pkgconfig/libtorrent-rasterbar.pc
The following lines have issues (specifically '//' in paths).
2:libdir=${prefix}//nix/store/l1730m3kbrp56csclwq2db69dsfi5lr5-libtorrent-rasterbar-2.0.11/lib
It is very likely that paths are being joined improperly.
ex: "${prefix}/@CMAKE_INSTALL_LIBDIR@" should be "@CMAKE_INSTALL_FULL_LIBDIR@"
Please see https://github.com/NixOS/nixpkgs/issues/144170 for more details.
```

so [check-pc-files-hook.sh](https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/cm/cmake/check-pc-files-hook.sh) works to detect broken paths like

```
libdir=${prefix}//nix/store/l1730m3kbrp56csclwq2db69dsfi5lr5-libtorrent-rasterbar-2.0.11/lib
```

but it fails to detect broken paths like

```
Cflags: ... -I/nix/store/yig4kl5kxz9njrhrklqaydpms14qqpqa-libtorrent-rasterbar-2.0.11//nix/store/xq5jkz4yv3aly4rl4ah3jfvmqvpv5c66-libtorrent-rasterbar-2.0.11-dev/include ...
```

because grep looks for `'}//nix/store'`

```
grepout=$(grep --line-number '}//nix/store' "$file" || true)
```

a generic autofixer could look like this

```
$ echo $'${prefix}//nix/store/x /nix/store/x//nix/store/x /nix/store/x//nix/store/x//nix/store/x' |
sed -E 's,(\$\{[^}]+\}|/nix/store/\S+)/(/nix/store/),\2,g'
/nix/store/x /nix/store/x /nix/store/x
```

ping maintainers
@nh2 @SuperSandro2000 @mweinelt @Artturin @r-burns @jtojnar @alexshpilkin @LunNova

via:
`git log -- pkgs/by-name/li/libtorrent-rasterbar-2_0_x/package.nix`
`git log -- pkgs/development/libraries/libtorrent-rasterbar/default.nix`
https://github.com/NixOS/nixpkgs/issues/144170
https://github.com/NixOS/nixpkgs/pull/18187

> CMake Error at torrent-rasterbar-pkgconfig/compile-settings-expanded.cmake:7:
>   Parse error.  Expected a newline, got identifier with text "set".

```diff
-        'set(_INSTALL_LIBDIR "@CMAKE_INSTALL_LIBDIR@") set(_INSTALL_FULL_LIBDIR "@CMAKE_INSTALL_FULL_LIBDIR@")'
+        'set(_INSTALL_LIBDIR "@CMAKE_INSTALL_LIBDIR@")'$'\n'""'set(_INSTALL_FULL_LIBDIR "@CMAKE_INSTALL_FULL_LIBDIR@")'
```

or

```diff
-        'set(_INSTALL_LIBDIR "@CMAKE_INSTALL_LIBDIR@") set(_INSTALL_FULL_LIBDIR "@CMAKE_INSTALL_FULL_LIBDIR@")'
+        "$(
+          echo 'set(_INSTALL_LIBDIR "@CMAKE_INSTALL_LIBDIR@")'
+          echo 'set(_INSTALL_FULL_LIBDIR "@CMAKE_INSTALL_FULL_LIBDIR@")'
+        )"
```

--- a/pkgs/by-name/li/libtorrent-rasterbar-2_0_x/package.nix
+++ b/pkgs/by-name/li/libtorrent-rasterbar-2_0_x/package.nix
@@ -55,6 +55,11 @@ stdenv.mkDerivation {
       --replace-fail 'set(_INSTALL_LIBDIR "@CMAKE_INSTALL_LIBDIR@")' \
                      'set(_INSTALL_LIBDIR "@CMAKE_INSTALL_LIBDIR@")
                       set(_INSTALL_FULL_LIBDIR "@CMAKE_INSTALL_FULL_LIBDIR@")'
+  ''
+  # a: libdir=''${prefix}//nix/store/...
+  # b: libdir=/nix/store/...
+  # https://github.com/NixOS/nixpkgs/issues/144170
+  + ''
     substituteInPlace cmake/Modules/GeneratePkgConfig/pkg-config.cmake.in \
       --replace-fail '$'{prefix}/@_INSTALL_LIBDIR@ @_INSTALL_FULL_LIBDIR@
   '';
@@ -67,6 +72,12 @@ stdenv.mkDerivation {
   postFixup = ''
     substituteInPlace "$dev/lib/cmake/LibtorrentRasterbar/LibtorrentRasterbarTargets-release.cmake" \
       --replace-fail "\''${_IMPORT_PREFIX}/lib" "$out/lib"
+  ''
+  # a: Cflags: ... -I/nix/store/x//nix/store/x-dev/include ...
+  # b: Cflags: ... -I/nix/store/x-dev/include ...
+  + ''
+    substituteInPlace $dev/lib/pkgconfig/libtorrent-rasterbar.pc \
+      --replace-fail "$out/$dev" "$dev"
   '';
 
   outputs = [

--- a/pkgs/by-name/li/libtorrent-rasterbar-2_0_x/package.nix
+++ b/pkgs/by-name/li/libtorrent-rasterbar-2_0_x/package.nix
@@ -1,35 +1,30 @@
 {
   lib,
   stdenv,
+  boost,
   fetchFromGitHub,
   cmake,
-  zlib,
-  boost,
   openssl,
   python3,
-  ncurses,
 }:
 
 let
-  version = "2.0.11";
-
   # Make sure we override python, so the correct version is chosen
   boostPython = boost.override {
     enablePython = true;
     python = python3;
   };
-
 in
-stdenv.mkDerivation {
+stdenv.mkDerivation (finalAttrs: {
   pname = "libtorrent-rasterbar";
-  inherit version;
+  version = "2.0.11";
 
   src = fetchFromGitHub {
     owner = "arvidn";
     repo = "libtorrent";
-    tag = "v${version}";
-    hash = "sha256-iph42iFEwP+lCWNPiOJJOejISFF6iwkGLY9Qg8J4tyo=";
+    tag = "v${finalAttrs.version}";
     fetchSubmodules = true;
+    hash = "sha256-iph42iFEwP+lCWNPiOJJOejISFF6iwkGLY9Qg8J4tyo=";
   };
 
   nativeBuildInputs = [
@@ -52,9 +47,10 @@ stdenv.mkDerivation {
   # https://github.com/arvidn/libtorrent/issues/6865
   postPatch = ''
     substituteInPlace cmake/Modules/GeneratePkgConfig/target-compile-settings.cmake.in \
-      --replace-fail 'set(_INSTALL_LIBDIR "@CMAKE_INSTALL_LIBDIR@")' \
-                     'set(_INSTALL_LIBDIR "@CMAKE_INSTALL_LIBDIR@")
-                      set(_INSTALL_FULL_LIBDIR "@CMAKE_INSTALL_FULL_LIBDIR@")'
+      --replace-fail \
+        'set(_INSTALL_LIBDIR "@CMAKE_INSTALL_LIBDIR@")' \
+        'set(_INSTALL_LIBDIR "@CMAKE_INSTALL_LIBDIR@")
+         set(_INSTALL_FULL_LIBDIR "@CMAKE_INSTALL_FULL_LIBDIR@")'
   ''
   # a: libdir=''${prefix}//nix/store/...
   # b: libdir=/nix/store/...
@@ -87,14 +83,15 @@ stdenv.mkDerivation {
   ];
 
   cmakeFlags = [
-    "-Dpython-bindings=on"
+    (lib.cmakeBool "python-bindings" true)
   ];
 
-  meta = with lib; {
+  meta = {
     homepage = "https://libtorrent.org/";
-    description = "C++ BitTorrent implementation focusing on efficiency and scalability";
-    license = licenses.bsd3;
+    description = "Efficient feature complete C++ bittorrent implementation";
+    changelog = "https://github.com/arvidn/libtorrent/blob/${finalAttrs.src.tag}/ChangeLog";
+    license = lib.licenses.bsd3;
     maintainers = [ ];
-    platforms = platforms.unix;
+    platforms = lib.platforms.unix;
   };
-}
+})

